From 45545589ab0d3f8d0a44752a5554e6a3d0e6dd9a Mon Sep 17 00:00:00 2001
From: Chaoyu Tang <ctang@cloudera.com>
Date: Wed, 8 Feb 2017 09:43:56 -0500
Subject: [PATCH 0973/1164] CDH-50072: HIVE-15815: Allow to pass some Oozie
 properties to Spark in HoS (Chaoyu Tang, reviewed
 by Xuefu Zhang)

Change-Id: Iae51ad5247dc03841718966bea957e7d5e38ef80
---
 .../hive/ql/exec/spark/HiveSparkClientFactory.java |    5 +++++
 1 file changed, 5 insertions(+)

diff --git a/ql/src/java/org/apache/hadoop/hive/ql/exec/spark/HiveSparkClientFactory.java b/ql/src/java/org/apache/hadoop/hive/ql/exec/spark/HiveSparkClientFactory.java
index f173719..f92e407 100644
--- a/ql/src/java/org/apache/hadoop/hive/ql/exec/spark/HiveSparkClientFactory.java
+++ b/ql/src/java/org/apache/hadoop/hive/ql/exec/spark/HiveSparkClientFactory.java
@@ -160,6 +160,11 @@ public static HiveSparkClient createHiveSparkClient(HiveConf hiveconf) throws Ex
         sparkConf.put("spark.hadoop." + propertyName, value);
         LOG.info(String.format(
           "load HBase configuration (%s -> %s).", propertyName, LogUtils.maskIfPassword(propertyName,value)));
+      } else if (propertyName.startsWith("oozie")) {
+        String value = hiveConf.get(propertyName);
+        sparkConf.put("spark." + propertyName, value);
+        LOG.info(String.format(
+          "Pass Oozie configuration (%s -> %s).", propertyName, LogUtils.maskIfPassword(propertyName,value)));
       }
 
       if (RpcConfiguration.HIVE_SPARK_RSC_CONFIGS.contains(propertyName)) {
-- 
1.7.9.5

